packages/@aws-cdk/aws-cognito-identitypool/lib/identitypool-user-pool-authentication-provider.ts

```diff
@@ -0,0 +1,120 @@
+import { randomBytes } from 'crypto';
+import {
+  CfnUserPool,
+  IUserPool,
+  UserPoolClient,
+  UserPoolClientProps,
+} from '@aws-cdk/aws-cognito';
+import {
+  Construct, Node,
+} from 'constructs';
+import { IIdentityPool } from './identitypool';
+
+/**
+ * Represents the concept of a User Pool Authentication Provider.
+ * You use user pool authentication providers to configure User Pools
+ * and User Pool Clients for use with Identity Pools
+ */
+export interface IUserPoolAuthenticationProvider {
+  /**
+   * The method called when a given User Pool Authentication Provider is added
+   * (for the first time) to an Identity Pool.
+   */
+  bind(
+    scope: Construct,
+    identityPool: IIdentityPool,
+    options?: UserPoolAuthenticationProviderBindOptions
+  ): UserPoolAuthenticationProviderBindConfig;
+}
+
+/**
+ * Props for the User Pool Authentication Provider
+ */
+export interface UserPoolAuthenticationProviderProps extends UserPoolClientProps, UserPoolAuthenticationProviderBindOptions {}
+
+/**
+ * Represents UserPoolAuthenticationProvider Bind Options
+ */
+export interface UserPoolAuthenticationProviderBindOptions {
```

Comment:
Let's just leave this empty for now and put `disableServerSideTokenCheck` in `UserPoolAuthenticationProviderProps`. Right now the one place we are using bind doesn't need to provide this.